streams/src/main/java/org/apache/kafka/streams/processor/StateStore.java

```diff
@@ -61,7 +84,14 @@
      * @throws IllegalStateException If store gets registered after initialized is already finished
      * @throws StreamsException if the store's change log does not contain the partition
      */
-    void init(ProcessorContext context, StateStore root);
+    default void init(final ProcessorContext<?, ?> context, final StateStore root) {
+        final org.apache.kafka.streams.processor.ProcessorContext adapted =
+            ProcessorContextReverseAdapter.adapt(
+                context,
+                new ProcessorContextReverseAdapter.UnsupportedDeprecatedForwarder()
+            );
+        init(adapted, root);
+    }
```

Comment:
We have to add this so that we can pass in the new ProcessorContext. The default implementation delegates to the old `init` method so that existing store implementations will function with no changes.

If the only callers were internal, we could just adapt at the call site. Unfortunately, users can also call `StateStore#init`, and they would do it if they have their own store implementations or if they use `MockProcessorContext` to test a stateful processor.